store/cachekv/store.go

```diff
@@ -325,6 +327,15 @@ func (store *Store) dirtyItems(start, end []byte) {
 		startIndex = 0
 	}
 
+	// since we took the time to sort the cache, we should use that effort
+	// we store at least THRESHOLD values -- cost of storing all is amortized across multiple calls
+	if endIndex-startIndex < THRESHOLD {
+		endIndex = math.MinInt(startIndex+THRESHOLD, len(strL)-1)
+		if endIndex-startIndex < THRESHOLD {
+			startIndex = math.MaxInt(endIndex-THRESHOLD, 0)
+		}
+	}
```

Comment:
Could you please explain this code more? I don't follow why specifically the span has to fit in < THRESHOLD and what this condition is? Thank you.

Reply:
The intention is that we actually remove some values from the unsorted cache instead of only a few or possibly none.  This code will extend start/end to at least THRESHOLD values.  If it's already above THRESHOLD, no changes are made.

I've updated the comment here, hopefully it's a bit clearer.  Let me know if you think it needs additional explanation.

Reply:
Thank you!